### ISSUE TITLE
`redundant_objc_attribute` false positives after updating 0.49.1->0.50.1

### ISSUE THREAD

**Issue by Cyberbeni** [realm/SwiftLint#4633]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

`redundant_objc_attribute` false positives after updating 0.49.1->0.50.1

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.50.1
* Installation method used (Homebrew, CocoaPods, building from source, etc)? pre-commit
* Paste your configuration file:

```yml
# insert yaml contents here
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
  If so, paste their relative paths and respective contents. No
* Which Xcode version are you using (check `xcodebuild -version`)? 14.0.1
* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

```swift
@objc public extension Foo {
  // This shouldn't trigger the warning
  @objc enum Bar: Int {
    case bar
  }
  // This will have compiler error without the @objc
  var bar: Bar { .bar }
  // This also triggers a warning and also causes compiler error but a better fix would be to mark other private interfaces @nonobjc
  @objc private enum Baz: Int {
    case baz
  }
  private var baz: Baz { .baz }
}
```

**Comment by markltownsend** [comment]
Any time frame on this PR being released? I also just updated to the latest SwiftLint and getting all these false positives that get compile errors if you fix the lint issues.
